src/store/useStore.ts

```diff
@@ -9,45 +9,27 @@
  *
  * All portions of the code written by Atypon Systems LLC are Copyright (c) 2019 Atypon Systems LLC. All Rights Reserved.
  */
-import { useLayoutEffect, useState } from 'react'
+
+import { useSyncExternalStoreWithSelector } from 'use-sync-external-store/with-selector'
 
 import deeperEqual from '../lib/deeper-equal'
 import { dispatch, GenericStore, state } from './Store'
 import { useGenericStore } from './StoreContext'
-// import { useSyncExternalStoreWithSelector } from 'use-sync-external-store/with-selector';
+
 type Selector<T> = (r: state) => state | T
+
 export const useStore = <T>(
-  selector?: Selector<T>
+  selector: Selector<T>
 ): [T, dispatch, () => state, GenericStore['subscribe']] => {
   const store = useGenericStore()
-  const init = selector ? () => selector(store.state!) : store.state
-  const [state, setState] = useState(init)
-
-  // @TODO - in react 18 these hooks usage will have to be replaced with this to work correctly with the concurrent rendering
-  //   const state = useSyncExternalStoreWithSelector(
-  //     store.subscribe,
-  //     store.getState(),
-  //     undefined,
-  //     selector,
-  //     deeperEqual
-  //   );
+  const init = selector ? () => selector(store.state!) : () => store.state
+  const state = useSyncExternalStoreWithSelector(
```

Comment:
did you test if it works well in terms of rerenders and selectors updates?

Reply:
good catch with this update though, thanks

Reply:
yes I have done a quick test and didn't notice any thing wrong.